lumen/sources/intake_sql.py

```diff
@@ -7,29 +9,72 @@ class IntakeBaseSQLSource(IntakeBaseSource):
     # Declare this source supports SQL transforms
     _supports_sql = True
 
-    @cached()
-    def get(self, table, **query):
-        '''
-        Applies SQL Transforms, creating new temp catalog on the fly
-        and querying the database.
-        '''
-        dask = query.pop('__dask', self.dask)
-        sql_transforms = query.pop('sql_transforms', [])
+    def _apply_transforms(self, source, sql_transforms):
+        if not sql_transforms:
+            return source
+        sql_expr = source._sql_expr
+        for sql_transform in sql_transforms:
+            sql_expr = sql_transform.apply(sql_expr)
+        return type(source)(**dict(source._init_args, sql_expr=sql_expr))
 
+    def _get_source(self, table):
         try:
             source = self.cat[table]
         except KeyError:
             raise KeyError(
                 f"'{table}' table could not be found in Intake catalog. "
                 f"Available tables include: {list(self.cat)}."
             )
-        sql_expr = source._sql_expr
-        for sql_transform in sql_transforms:
-            sql_expr = sql_transform.apply(sql_expr)
-        new_source = type(source)(**dict(source._init_args, sql_expr=sql_expr))
-        df = self._read(new_source)
+        return source
+
+    @cached()
+    def get(self, table, **query):
+        '''
+        Applies SQL Transforms, creating new temp catalog on the fly
+        and querying the database.
+        '''
+        dask = query.pop('__dask', self.dask)
+        sql_transforms = query.pop('sql_transforms', [])
+        source = self._get_source(table)
+        source = self._apply_transforms(source, sql_transforms)
+        df = self._read(source)
         return df if dask or not hasattr(df, 'compute') else df.compute()
 
+    def get_schema(self, table=None):
+        if table is None:
+            tables = self.get_tables()
+        else:
+            tables = [table]
+
+        schemas = {}
+        limit = SQLLimit(limit=1)
+        for table in tables:
+            if not self.load_schema:
+                schemas[entry] = {}
+                continue
+            source = self._get_source(table)
+            data = self._read(self._apply_transforms(source, [limit]))
+            schema = get_dataframe_schema(data)['items']['properties']
+            enums, min_maxes = [], []
```

Comment:
Edge cases that could possibly come up
1. What happens with an ID field that is a string? There could be an enum that is millions of distinct digits but in reality we don't want as a drop down or as part of anything on the UI
2. What happens if there is a numeric field we want to be an enum in the source data? (Product 1, 2, 3, etc...) (Could we get around this by casting as a string type before reading from source if the user wants to or specify somewhere here?)
3. What about boolean field values?

Reply:
Maybe set a cap of how many unique values an enum can have?

Reply:
> Maybe set a cap of how many unique values an enum can have?

Yes, that seems like the best approach, just append a LIMIT 1000 SQLTransform.

Reply:
Thanks Philipp, on 3 makes sense, on 2 we can append a space or word to the number as a new column and just use that as the enum.  For 1, not sure how LIMIT 1000 would work, but something like that would be great so it doesn't blow up but we can still reference it in later transforms.